third_party/terraform/resources/resource_google_project_default_service_accounts.go

```diff
@@ -77,7 +78,13 @@ func resourceGoogleProjectDefaultServiceAccountsDoAction(d *schema.ResourceData,
 		}
 	case "UNDELETE":
 		_, err := config.NewIamClient(userAgent).Projects.ServiceAccounts.Undelete(serviceAccountSelfLink, &iam.UndeleteServiceAccountRequest{}).Do()
-		if err != nil {
+		errExpected := restorePolicy == "REVERT_AND_IGNORE_FAILURE"
+		errReceived := err != nil
+		if errExpected && errReceived {
```

Comment:
probably not worth the round trip to change, but this would be clearer as

```
if errReceived {
  if !errExpected {
    return error here
  }

  log here
}
```

Reply:
I'll keep in mind for the next iteration but I thought this is a common practice in Go code, mainly on tests.